File: medium/48/rotate_image.cpp
#include <algorithm>

class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        int i, j;
        i = 0;
        j = matrix.size() - 1;
        
        while (i < j) {
            for (int p = j - i; p > 0; p--) {
                std::swap(matrix[i][i + p], matrix[j][j - p]);
                std::swap(matrix[j][j - p], matrix[i + p][j]);
                std::swap(matrix[i][i + p], matrix[j - p][i]);
            }
            i++;
            j--;
        }
    }
};

File: medium/48/rotate_image.py

class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        i = 0
        j = len(matrix) - 1

        while i < j:
            for p in range(j-i):
                matrix[i][i + p], matrix[j - p][i], matrix[j][j - p], matrix[i + p][j] = \
                matrix[j - p][i], matrix[j][j - p], matrix[i + p][j], matrix[i][i + p]

            i += 1
            j -= 1
